homeassistant/components/intellifire/fan.py

```diff
@@ -0,0 +1,156 @@
+) -> None:
+    """Set up the fans."""
+    coordinator: IntellifireDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+
+    if coordinator.data.has_fan:
+        async_add_entities(
+            IntellifireFan(coordinator=coordinator, description=description)
+            for description in INTELLIFIRE_FANS
+        )
+        return
+    LOGGER.info("Disabling Fan - IntelliFire device does not appear to have one")
+
+
+class IntellifireFan(IntellifireEntity, FanEntity):
+    """This is Fan entity for the fireplace."""
+
+    entity_description: IntellifireFanEntityDescription
+    _attr_supported_features = FanEntityFeature.SET_SPEED | FanEntityFeature.PRESET_MODE
+
+    @property
+    def is_on(self):
+        """Return on or off."""
+        return self.entity_description.value_fn(self.coordinator.read_api.data) >= 1
+
+    @property
+    def percentage(self) -> int | None:
+        """Return fan percentage."""
+        percent_step = ordered_list_item_to_percentage(
+            self.entity_description.named_speeds,
+            self.entity_description.named_speeds[0],
+        )
+        return (
+            self.entity_description.value_fn(self.coordinator.read_api.data)
+            * percent_step
+        )
+
+    @property
+    def speed_count(self) -> int:
+        """Count of supported speeds."""
+        return len(self.entity_description.named_speeds)
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set the preset mode of the fan."""
+        # Get a percent from the preset
+
+        percent = ordered_list_item_to_percentage(
+            self.entity_description.named_speeds, preset_mode
+        )
+        await self.async_set_percentage(percent)
+
+    async def async_set_percentage(self, percentage: int) -> None:
+        """Set the speed percentage of the fan."""
+        # Calculate percentage steps
+        LOGGER.debug("Setting Fan Speed %s", percentage)
+        percent_step = 100.0 / len(self.entity_description.named_speeds)
+        int_value = int(math.ceil(float(percentage) / percent_step))
+        await self.entity_description.set_fn(self.coordinator.control_api, int_value)
+        setattr(
+            self.coordinator.read_api, self.entity_description.data_field, int_value
+        )
+        self.async_write_ha_state()
+        LOGGER.debug(
+            "Fan speed %s [%s%%] = %s: [%s]",
+            int_value,
+            int_value * 25,
+            self.entity_description.name,
+            percentage_to_ordered_list_item(
+                self.entity_description.named_speeds, int(int_value * percent_step)
+            ),
+        )
+
+    async def async_turn_on(
+        self,
+        percentage: int | None = None,
+        preset_mode: str | None = None,
+        **kwargs: Any,
+    ) -> None:
+        """Turn on the fan."""
+        LOGGER.debug("%s,%s", percentage, preset_mode)
+        LOGGER.debug("On  [ Set Function   ] - Turn On Fan")
```

Comment:
Please combine these message us similar formatting to other debug messages in Home Assistant

Reply:
Same for the rest of the logger messages